### PR TITLE
#1023 Add support for horizontal lines on wig tracks

### PR DESCRIPTION
**Pull request by docwarren**
@jrobinso 
#1023
I squashed all of the commits into a single commit and made a new pull request for this one, so hopefully its easier for you to see what I've done.

Here is a breakdown:
- simplify the maths of calculating y and height when rendering
- abstract out methods for getting height, y and x
- moving error catching into abstracted methods
- removing unused arguments
- changing this to self
- add method for drawing horizontal lines
- simplify dashed line function in igv-canvas
- add example files for manual testing of wig tracks

The following numbered changes are from merging in master changes. (i.e. they are not changes that I made)
11
50
8
11
89 - featureFIleReader
49

The following are changes I made:
13
89 - wigs.html
89 - wigTrack.js
33 - igv-canvas

### CODE REVIEW

**Comment by jrobinso** [comment]
OK thanks.  I'll get to this soon.

**Comment by docwarren** [comment]
No rush. Thanks.

**Comment by jrobinso** [comment]
Hi,  back and ready to merge.  We need to update the wiki doc  for the new feature(s),  you can't easily do that with a PR so could you summarize what needs added here as a comment.  Specifically we need to update this:  https://github.com/igvteam/igv.js/wiki/Wig-Track

**Comment by docwarren** [comment]
Great. Thanks!

**Comment by docwarren** [comment]
<html>
<table>
  <thead>
    <tr>
      <th>Property</th>
      <th>Description</th>
      <th>Default</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>autoscale</td>
      <td>Autoscale track to maximum value in view</td>
      <td></td>
    </tr>
    <tr>
      <td>autoscaleGroup</td>
      <td>Identifier for an autoscale group. Tracks with the same identifier are autoscaled together. </td>
    </tr>
    <tr>
      <td>min</td>
      <td>Sets the minimum value for the data (y-axis) scale. Usually zero.</td>
      <td>0</td>
    </tr>
    <tr>
      <td>max</td>
      <td>Sets the maximum value for the data (y-axis) scale. This value is ignored if autoscale = true</td>
      <td>No default</td>
    </tr>
    <tr>
      <td>color</td>
      <td>Track color.</td>
      <td>"rgb(150,150,150)"</td>
    </tr>
    <tr>
      <td>guideLines</td>
      <td>
        Draws a horizontal line for each object in the given array.</br>
        guideLines: [ {color: [color], y: [number], dotted: [bool]} ]</br>
        y value should be between min and max or it will not show.
      </td>
      <td></td>
    </tr>
  </tbody>

 
Example
```
{
  type: "wig",
  name: "CTCF",
  url: "https://www.encodeproject.org/files/ENCFF356YES/@@download/ENCFF356YES.bigWig",
  min: "0",
  max: "30",
  color: "rgb(0, 0, 150)",
  guideLines: [
    {color: 'green', dotted: true, y: 25}, 
    {color: 'red', dotted: false, y: 5}
  ]
}
```
